lib/charms/mongodb/v0/mongos.py

```diff
@@ -158,6 +166,75 @@ def add_shard(self, shard_name, shard_hosts, shard_port=Config.MONGODB_PORT):
         logger.info("Adding shard %s", shard_name)
         self.client.admin.command("addShard", shard_url)
 
+    def remove_shard(self, shard_name: str) -> None:
+        """Removes shard from the cluster.
+
+        Raises:
+            ConfigurationError, OperationFailure, NotReadyError,
+            RemovePrimaryShardError
+        """
+        sc_status = self.client.admin.command("listShards")
+        # It is necessary to call removeShard multiple times on a shard to guarantee removal.
+        # Allow re-removal of shards that are currently draining.
+        if self._is_any_draining(sc_status, ignore_shard=shard_name):
+            cannot_remove_shard = (
+                f"cannot remove shard {shard_name} from cluster, another shard is draining"
+            )
+            logger.info(cannot_remove_shard)
+            raise NotReadyError(cannot_remove_shard)
+
+        # TODO Follow up PR, there is no MongoDB command to retrieve primary shard, this is
+        # possible with mongosh.
+        primary_shard = False
+        if primary_shard:
+            # TODO Future PR, support removing Primary Shard if there are no unsharded collections
+            # on it. All sharded collections should perform `MovePrimary`
+            cannot_remove_primary_shard = (
+                f"Shard {shard_name} is the primary shard, cannot remove."
+            )
+            logger.error(cannot_remove_primary_shard)
+            raise RemovePrimaryShardError(cannot_remove_primary_shard)
+
+        logger.info("Attempting to remove shard %s", shard_name)
+        self.client.admin.command("removeShard", shard_name)
+        logger.info("Shard %s, now draining", shard_name)
```

Comment:
Please extend the log with `remaining.chunks` and `dbsToMove` info from `removeShard` output.
e.g.
```suggestion
        logger.info("Shard %s is %s now. Remaining chunks: %i. DBs to move: %s.", shard_name, status["state"], status["remaining"]["chunks"], ",".join(status["dbsToMove"])
```